docs: add IDKit invite-code mode page (#111)
worldcoin/developer-docs: world-id/idkit/invite-code.mdx

---
title: "Invite-code mode"
description: "Onboard users into the World Network with a short 6-character invite code instead of a QR scan."
"og:image": "https://raw.githubusercontent.com/worldcoin/developer-docs/main/images/docs/docs-meta.png"
"twitter:image": "https://raw.githubusercontent.com/worldcoin/developer-docs/main/images/docs/docs-meta.png"
---

Invite-code mode displays a short 6-character code in your app that the user enters into World App. World App treats the code as an entry point to the in-app onboarding flows the user needs to complete in order to satisfy your IDKit request, then returns the proof.

<Note>

>   Invite-code mode requires World App on iOS. 

Author: Probably worth a note that only selfie-check is supported atm. As we add the other credentials we can update the docs.

>   Invite-code mode currently requires World App on iOS. Users on Android will not be able to redeem the code today.

Author: ```suggestion
  Invite-code mode requires World App on iOS. 
```

---
title: "Invite-code mode"
description: "Onboard users into the World Network with a short 6-character invite code instead of a QR scan."
"og:image": "https://raw.githubusercontent.com/worldcoin/developer-docs/main/images/docs/docs-meta.png"
"twitter:image": "https://raw.githubusercontent.com/worldcoin/developer-docs/main/images/docs/docs-meta.png"
---

Invite-code mode displays a short 6-character code in your app that the user enters into World App. World App treats the code as an entry point to the in-app onboarding flows the user needs to complete in order to satisfy your IDKit request, then returns the proof.

<Note>

>   Invite-code mode requires World App on iOS. 

Author: ```suggestion
  Invite-code mode is only supported by World App on iOS. 
```

</Note>

<Note>

>   Only the `selfieCheckLegacy` preset is supported today. The docs will be updated as support is added for other credentials.

Author: ```suggestion
  Only the `selfieCheckLegacy` preset is supported today. 
```

</Note>

## When to use it

> Use invite-code mode to expedite onboarding users who don't yet hold the credential you're requesting — World App walks them through getting it. It also fits cross-device flows, where typing a short code is more robust than scanning a QR.

Author: ```suggestion
Use invite-code mode to expedite onboarding users who don't yet hold the credential you're requesting — World App walks them through getting it. It also fits cross-device flows, where typing a short code can be easier than scanning a QR.
```

---
title: "Invite-code mode"
description: "Onboard users into the World Network with a short 6-character invite code instead of a QR scan."
"og:image": "https://raw.githubusercontent.com/worldcoin/developer-docs/main/images/docs/docs-meta.png"
"twitter:image": "https://raw.githubusercontent.com/worldcoin/developer-docs/main/images/docs/docs-meta.png"
---

Invite-code mode is a sibling to the default QR / connect-URL flow. Instead of showing a QR code that the user scans with World App, your app displays a short **6-character code** that the user enters into World App.

The proof, polling lifecycle, status enum, and verification step are identical to QR mode. Only the discovery channel changes.

## When to use it

> Invite-code mode is built to **expedite onboarding users into the World Network**. When a user enters your app's invite code into World App, World App uses it to launch the in-app onboarding flows the user needs to complete in order to satisfy your IDKit request. This lets you request a credential the user does not yet hold — World App walks them through getting it, then returns the proof to your app.

Author: Add a note that this is for iOS only

---
title: "Invite-code mode"
description: "Verify users with World ID across devices using a 6-character invite code instead of a QR scan."
"og:image": "https://raw.githubusercontent.com/worldcoin/developer-docs/main/images/docs/docs-meta.png"
"twitter:image": "https://raw.githubusercontent.com/worldcoin/developer-docs/main/images/docs/docs-meta.png"
---

Invite-code mode is a sibling to the default QR / connect-URL flow. Instead of showing a QR code that the user scans with World App, your app displays a short **6-character code** that the user types into World App on a different device.

The proof, polling lifecycle, status enum, and verification step are identical to QR mode. Only the discovery channel changes.

## When to use it

> Use invite-code mode when the relying party (your app) and World App are reliably **on different devices** — for example, a desktop browser experience where the user has World App on their phone. Typing a 6-character code is more robust than asking the user to point a phone camera at their own laptop screen.

Author: This isn't the reason we built it, but is something we ensured worked. The reason to use this feature is to expedite onboarding users into the World Network. The app uses the code to show in-app flows that users complete as part of onboarding to respond to the IDKit request.

- World App accepts the code in upper or lower case and ignores separators when the user types it in.

## Lifecycle

- The bridge issues the code with a short TTL (currently ten minutes). Surface the expiry to the user with a countdown.
- The code is **one-shot**: once redeemed in World App, it cannot be reused.
- After the user redeems the code, your app's existing poll loop receives the proof exactly as it does in QR mode.

## Integrate

The setup steps that precede the request — creating an app in the Developer Portal and generating an RP signature on your backend — are identical to the [standard integration](/world-id/idkit/integrate). Only the request call and the rendered UI change.

### Create the request

<CodeGroup title="Create an invite-code request">
```typescript title="JavaScript"
import { IDKit, orbLegacy } from "@worldcoin/idkit-core";

const request = await IDKit.requestWithInviteCode({
  app_id: "app_xxxxx",
  action: "my-action",
  rp_context: {
    rp_id: "rp_xxxxx",
    nonce: rpSig.nonce,
    created_at: rpSig.created_at,
    expires_at: rpSig.expires_at,
    signature: rpSig.sig,
  },
  allow_legacy_proofs: true,
  environment: "production",

> }).preset(orbLegacy({ signal: "user-123" }));

Author: Let's put the selfie check preset here since that's the hero use case for this atm.

  onOpenChange={setOpen}
  app_id="app_xxxxx"
  action="my-action"
  rp_context={rpContext}
  allow_legacy_proofs={true}
  preset={orbLegacy({ signal: "user-123" })}
  handleVerify={async (result) => {
    const response = await fetch("/api/verify-proof", {
      method: "POST",
      headers: { "content-type": "application/json" },
      body: JSON.stringify({ rp_id: rpContext.rp_id, idkitResponse: result }),
    });
    if (!response.ok) throw new Error("Backend verification failed");
  }}
  onSuccess={() => {
    // Update app state here.
  }}
/>;
```

```swift title="Swift"
import IDKit

let config = IDKitRequestConfig(
  appId: "app_xxxxx",
  action: "my-action",
  rpContext: rpContext,
  allowLegacyProofs: true,
  environment: .production
)

let request = try IDKit.request(config: config)
  .presetWithInviteCode(orbLegacy(signal: "user-123"))

let code = request.code              // 6-character canonical form
let expiresAt = request.expiresAt    // Date

let completion = await request.pollUntilCompletion()
```
</CodeGroup>

The config object is the same one you pass to `IDKit.request(...)` — invite-code mode introduces no new required fields.

### Display the code

> Show `code` prominently. A formatted display (`ABC-DEF` in a large, monospaced font) is easier to read aloud and harder to mistype than the unseparated form. The Next.js example app and iOS sample app in the IDKit repo both ship a reference UI you can crib from.

Author: Remove comments on reading the code aloud.

Author: "Crib from" is unprofessional - "can be referenced" is better, but we're also going to provide at least a react component so this shouldn't be really mentioned.
